snuba/api.py

```diff
@@ -262,25 +269,32 @@ def json_default(obj):
 
 
 @split_query
-def parse_and_run_query(dataset, request: Request, timer):
-    from_date, to_date = TimeSeriesExtensionProcessor.get_time_limit(request.extensions['timeseries'])
+def parse_and_run_query(
+    dataset,
+    request: Request,
+    extension_payloads: Mapping[str, Mapping[str, Any]],
+    timer,
+):
+    from_date, to_date = TimeSeriesExtensionProcessor.get_time_limit(
+        TimeSeriesExtension.parse_payload(extension_payloads['timeseries'])
+    )
 
     extensions = dataset.get_extensions()
     for name, extension in extensions.items():
         extension.get_processor().process_query(
             request.query,
-            request.extensions[name],
+            extension.parse_payload(extension_payloads[name]),
```

Comment:
I think `parse_payload` should be a private thing in the extension rather than exposed like this.